phpBB/phpbb/search/backend/search_backend_interface.php

```diff
@@ -0,0 +1,155 @@
+	 * Returns the search_query
+	 *
+	 * @return string search query
+	 */
+	public function get_search_query();
+
+	/**
+	 * Returns the common_words array
+	 *
+	 * @return array common words that are ignored by search backend
+	 */
+	public function get_common_words();
+
+	/**
+	 * Returns the word_length array
+	 *
+	 * @return array min and max word length for searching
+	 */
+	public function get_word_length();
+
+	//public function init();
+
+	/**
+	* Splits keywords entered by a user into an array of words stored in $this->split_words
+	* Stores the tidied search query in $this->search_query
+	*
+	* @param string &$keywords Contains the keyword as entered by the user
+	* @param string $terms is either 'all' or 'any'
+	* @return bool false if no valid keywords were found and otherwise true
+	*/
+	public function split_keywords(&$keywords, $terms);
+
+	/**
+	* Performs a search on keywords depending on display specific params. You have to run split_keywords() first
+	*
+	* @param	string		$type				contains either posts or topics depending on what should be searched for
+	* @param	string		$fields				contains either titleonly (topic titles should be searched), msgonly (only message bodies should be searched), firstpost (only subject and body of the first post should be searched) or all (all post bodies and subjects should be searched)
+	* @param	string		$terms				is either 'all' (use query as entered, words without prefix should default to "have to be in field") or 'any' (ignore search query parts and just return all posts that contain any of the specified words)
+	* @param	array		$sort_by_sql		contains SQL code for the ORDER BY part of a query
+	* @param	string		$sort_key			is the key of $sort_by_sql for the selected sorting
+	* @param	string		$sort_dir			is either a or d representing ASC and DESC
+	* @param	string		$sort_days			specifies the maximum amount of days a post may be old
+	* @param	array		$ex_fid_ary			specifies an array of forum ids which should not be searched
+	* @param	string		$post_visibility	specifies which types of posts the user can view in which forums
+	* @param	int			$topic_id			is set to 0 or a topic id, if it is not 0 then only posts in this topic should be searched
+	* @param	array		$author_ary			an array of author ids if the author should be ignored during the search the array is empty
+	* @param	string		$author_name		specifies the author match, when ANONYMOUS is also a search-match
+	* @param	array		&$id_ary			passed by reference, to be filled with ids for the page specified by $start and $per_page, should be ordered
+	* @param	int			$start				indicates the first index of the page
+	* @param	int			$per_page			number of ids each page is supposed to contain
+	* @return	boolean|int						total number of results
+	*/
+	public function keyword_search($type, $fields, $terms, $sort_by_sql, $sort_key, $sort_dir, $sort_days, $ex_fid_ary, $post_visibility, $topic_id, $author_ary, $author_name, &$id_ary, &$start, $per_page);
```

Comment:
Same here. As we're using PHP7+ only we should opt for specifying parameter types in new code.